library/Vanilla/Models/SSOInfo.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * @author Alexandre (DaazKu) Chouinard <alexandre.c@vanillaforums.com>
+ * @copyright 2009-2017 Vanilla Forums Inc.
+ * @license http://www.opensource.org/licenses/gpl-2.0.php GNU GPL v2
+ */
+
+namespace Vanilla\Models;
+
+use Vanilla\ArrayAccessTrait;
+
+/**
+ * Class SSOInfo
+ *
+ * This is a data object containing information returned by an SSOAuthenticator.
+ * Adding methods to this object probably means that you are doing something wrong.
+ *
+ * @package Vanilla\Models
+ */
+class SSOInfo implements \ArrayAccess {
+    use ArrayAccessTrait;
+
+    /**
+     * Maps to "GDN_UserAuthenticationProvider.AuthenticationSchemeAlias"
+     *
+     * @var string
+     */
+    public $authenticatorName;
+
+    /**
+     * Maps to "GDN_UserAuthenticationProvider.ProviderKey"
+     *
+     * @var string
+     */
+    public $authenticatorID;
+
+    /**
+     * Whether the authenticator can sync Roles or User info.
+     *
+     * @var bool
+     */
+    public $authenticatorIsTrusted;
+
+    /**
+     * Maps to "GDN_UserAuthentication.ForeignUserKey"
+     *
+     * @var string
+     */
+    public $uniqueID;
+
+    /**
+     * SSOInfo constructor.
+     *
+     * @param array $associativeArray
+     * @throws \Exception If a non associative array is passed to the constructor.
+     */
+    public function __construct(array $associativeArray = []) {
+        if (array_key_exists(0, $associativeArray)) {
+            throw new \Exception(__CLASS__.' can only be initialized with an associative array.');
+        }
+
+        foreach($associativeArray as $name => $value) {
+            $this->$name = $value;
```

Comment:
Instead of dynamically declaring a bunch of properties on your object, how about keeping them in a dedicated data array that you define on the class? I know it probably won't mesh well with your intentions for using `ArrayAccessTrait`, but I think we should stay away from dynamically-declared properties, especially when they're arbitrary.

Reply:
That's actually how I did it first but that's a no go. This is an array like object and the fact that it is an object should be transparent.

Due to the fact that this is an array like object, dynamically declared properties is perfectly fine.